assets/theme-editor.js

```diff
@@ -1,4 +1,10 @@
+const hideProductModal = () => {
+  const productModal = document.querySelectorAll('product-modal[open]');
+  productModal && productModal.forEach(modal => modal.hide());
+}
+
 document.addEventListener('shopify:block:select', function(event) {
+  hideProductModal();
   const blockSelectedIsSlide = event.target.classList.contains('slideshow__slide');
   if (!blockSelectedIsSlide) return;
 
```

Comment:
I think we could maybe apply to `shopify:block:deselect` and `shopify:section:load` 🤔 
What do you think ? 

[video for block deselect](https://screenshot.click/22-34-m9qu6-hnbst.mp4), not sure it's really needed there. 

But I think it makes sense to add when adding a section: [video](https://screenshot.click/22-38-9yx3f-4cmz2.mp4)